File: Java/src/BaekJoon/no34472_교도소/Main2.java
package BaekJoon.no34472_교도소;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main2 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] input = br.readLine().split(" ");
        long[] rooms = new long[n];
        long avg = 0L, ans = 0L;

        for (int i = 0; i < n; i++) {
            rooms[i] = Long.parseLong(input[i]);
            avg += rooms[i];
        }

        avg /= n;
        for (int i = 0; i < 2 * n - 1; i++) {
            int cur = i % n, next = (i + 1) % n;

            if (rooms[cur] > avg) {
                long move = rooms[cur] - avg;

                ans += move;
                rooms[cur] -= move;
                rooms[next] += move;
            }
        }

        System.out.println(ans);
    }
}

File: Kotlin/src/main/kotlin/BaekJoon/no25418_정수a를k로만들기/Main2.kt
package BaekJoon.no25418_정수a를k로만들기

fun main() = with(System.`in`.bufferedReader()) {
    var (op, ed) = readLine().split(" ").map { it.toInt() };

    var ans = 0;
    while (ed != op) {
        if (ed % 2 == 0 && ed / 2 >= op) ed /= 2 else ed -= 1;
        ++ans;
    }
    print(ans);
}